SU2_CFD/include/fluid/CFluidFlamelet.hpp

```diff
@@ -0,0 +1,127 @@
+  unsigned short n_control_vars; /*!< \brief number of controlling variables. */
+
+  vector<string> table_scalar_names; /*!< \brief vector to store names of scalar variables.   */
+  vector<string> table_lookup_names; /*!< \brief vector to store names of look up variables.   */
+
+  su2double mass_diffusivity; /*!< \brief local mass diffusivity of the mixture */
+  su2double molar_weight;     /*!< \brief local molar weight of the mixture */
+
+  CLookUpTable* look_up_table;
+
+  vector<string> varnames_TD;     /*!< \brief Lookup names for thermodynamic state variables. */
+  vector<su2double> val_vars_TD; /*!< \brief References to thermodynamic state variables. */
+
+ public:
+  CFluidFlamelet(CConfig* config, su2double value_pressure_operating);
+
+  ~CFluidFlamelet();
+
+  /*!
+   * \brief Set the thermodynamioc state.
+   * \param[in] val_temperature - temperature
+   * \param[in] val_scalars - pointer to species mass fractions
+   */
+  void SetTDState_T(su2double val_temperature, const su2double* val_scalars = nullptr) override;
+
+  /*!
+   * \brief Get the total enthalpy from the tabulated temperature and species (inverse lookup).
+   * \param[in/out] enthalpy - total enthalpy
+   * \param[in] val_prog - progress variable
+   * \param[in] val_temp - temperature
+   * \param[in] initial_value - initial value for the iterative lookup procedure.
+   * \param[out] exit_code = error code
+   */
+  unsigned long GetEnthFromTemp(su2double& enthalpy, const su2double val_prog, const su2double val_temp,
+                                su2double initial_value = 0);
+
+  /*!
+   * \brief Return a pointer to the lookup table
+   * \param[out] look_up_table - pointer to lookup table
+   */
+  inline CLookUpTable* GetLookUpTable() { return look_up_table; }
+
+  /*!
+   * \brief Get the mass diffusivity of the species.
+   * \param[in] iVar - index to the species
+   * \param[out] mass_diffusivity - value of the mass diffusivity
+   */
+  inline su2double GetMassDiffusivity(int iVar) final { return mass_diffusivity; }
+
+  /*!
+   * \brief Get the thermal conductivity of the species.
+   * \param[in] iVar - index to the species
+   * \param[out] Kt - value of the thermal conductivity
+   */
+  inline su2double GetThermalConductivity() { return Kt; }
+
+  /*!
+   * \brief Get the laminar viscosity of the species.
+   * \param[in] iVar - index to the species
+   * \param[out] Mu - value of the laminar viscosity
+   */
+  inline su2double GetLaminarViscosity() { return Mu; }
+
+  /*!
+   * \brief Get the index of a string in a vector.
+   */
+  inline int GetIdx(vector<string> v,string s) {
+
+    auto it = find(v.begin(), v.end(), s);
+  
+    if (it != v.end()) 
+        return(it - v.begin());
+    else 
+        return(-1);
+  }
+
+  /*!
+   * \brief Get the value of the LUT variable using the string name.
+   */
+  inline su2double GetLUTVal(string varname_TD) {
+      return val_vars_TD[GetIdx(varnames_TD,varname_TD)];
+    };
```

Comment:
Same as above
```suggestion
  inline su2double GetLUTVal(string varname_TD) {
      return val_vars_TD[GetIdx(varnames_TD,varname_TD)];
    }
```

Reply:
OK: Private, pass args by const reference, make function const or static, run clang-format on new files (formatting is off here).